src/main/java/org/broadinstitute/hellbender/utils/config/ConfigUtils.java

```diff
@@ -0,0 +1,332 @@
+                        }
+                    }
+                }
+            }
+
+            // Grab the system properties:
+            Properties systemProperties = System.getProperties();
+
+            // Grab the environment properties:
+            Map<String, String> environmentProperties = System.getenv();
+
+            // Make sure that if our property isn't in the system and environment
+            // properties, that we set it to a neutral value that will not contain
+            // anything (so that the property will fall back into the next value).
+            for (String property : propertyNames) {
+
+                if ((!environmentProperties.keySet().contains(property)) &&
+                        (!systemProperties.containsKey(property))) {
+
+                    ConfigFactory.setProperty(property, "/dev/null");
+                }
+            }
+
+            hasSetConfigFactoryVariableDefaults = true;
+        }
+    }
+
+    /**
+     * Get the configuration file name from the given arguments.
+     * Modifies the given arguments to remove both the configuration file specification string
+     * and the configuration file name from the args.
+     *
+     * NOTE: Does NOT validate that the resulting string is a valid configuration file.
+     *
+     * @param args Command-line arguments passed to this program.
+     * @param configFileOption The command-line option indicating that the config file is next
+     * @return The name of the configuration file for this program or {@code null}.
+     */
+    public static final String getConfigFilenameFromArgs( final ArrayList<String> args, final String configFileOption ) {
+
+        Utils.nonNull(args);
+        Utils.nonNull(configFileOption);
+
+        String configFileName = null;
+
+        for ( int i = 0 ; i < args.size() ; ++i ) {
+            if (args.get(i).compareTo(configFileOption) == 0) {
+
+                // Get rid of the command-line argument name:
+                args.remove(i);
+
+                if ( i < args.size() ) {
+
+                    // Get and remove the specified config file:
+                    configFileName = args.remove(i);
+                    break;
+                }
+                else {
+                    // Option was provided, but no file was specified.
+                    // We cannot work under these conditions:
+
+                    String message = "ERROR: Configuration file not given after config file option specified: " + configFileOption;
+                    System.err.println(message);
+                    throw new UserException.BadInput(message);
+                }
+            }
+        }
+
+        return configFileName;
+    }
+
+    /**
+     * Get the given key from a {@link GATKConfig} class.
+     * @param key The key in the {@link GATKConfig} to get.
+     * @param <T> The type of the key to obtain.
+     * @return The value for the specified {@code key}.
+     */
+    @SuppressWarnings("unchecked")
```

Comment:
Interesting.  I didn't realize you can do that.

I actually removed this whole method because it allows for some really bad practices and would be detrimental to the codebase to keep it there.